ros_ws/src/car_control/src/car_control.cpp

```diff
@@ -0,0 +1,63 @@
+#include "car_control.h"
+
+CarControl::CarControl()
+    : run{ true }
+    , speed{ 0 }
+    , angle{ 0 }
+{
+    in_drive_param = nh_.subscribe< drive_msgs::drive_param >(
+        TOPIC_DRIVE_PARAM, 1, &CarControl::drive_param_callback, this);
+    in_cmd = nh_.subscribe< std_msgs::String >(TOPIC_CMD, 1,
+                                               &CarControl::cmd_callback, this);
+
+    out_speed = nh_.advertise< std_msgs::Float64 >(TOPIC_FOCBOX_SPEED, 1);
+    out_angle = nh_.advertise< std_msgs::Float64 >(TOPIC_FOCBOX_ANGLE, 1);
+}
+
+void CarControl::drive_param_callback(
+    const drive_msgs::drive_param::ConstPtr& param)
+{
+    adjustDriveParam(param->velocity, param->angle);
+}
+
+void CarControl::adjustDriveParam(double raw_speed, double raw_angle)
+{
+    speed = raw_speed * MAX_SPEED;
+    if (speed < MIN_SPEED)
+    {
+        speed = 0;
+    }
+    angle = (raw_angle * MAX_ANGLE + 1) / 2;
+    std::cout << "speed: " << speed << " | angle: " << angle << std::endl;
+    if (run)
+    {
+        std_msgs::Float64 msg_speed;
+        msg_speed.data = speed;
+        out_speed.publish(msg_speed);
+        std_msgs::Float64 msg;
+        msg.data = angle;
+        out_angle.publish(msg);
+    }
+}
+
+void CarControl::cmd_callback(const std_msgs::String::ConstPtr& cmd)
+{
+    std::string str = cmd->data;
```

Comment:
Please rename `str` to something that explains what the variable does.

Reply:
done